### PR TITLE
docs(OS-661): moves docs for the developer portal from `docs/osx` to `docs/developer-portal`

### PR DESCRIPTION
**Pull request by mathewmeconry**
## Description

Please include a summary of the change and be sure you follow the contributions rules we do provide [here](./CONTRIBUTIONS.md)

Task: OS-661

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [x] I have updated the `CHANGELOG.md` file in the root folder.
- [x] I have updated the `DEPLOYMENT_CHECKLIST` file in the root folder.
- [x] I have updated the `UPDATE_CHECKLIST` file in the root folder.

### CODE REVIEW

File: packages/contracts/docs/osx/03-reference-guide/index.md [hunk 1]
```diff
@@ -1,7 +0,0 @@
----
```
Comment (Michael-A-Heuer) [inline packages/contracts/docs/osx/03-reference-guide/index.md:1]: This file should not be deleted but moved to `docs/developer-portal`

File: .github/workflows/documentation-update.yml [hunk 2]
```diff
@@ -26,7 +26,7 @@ jobs:
         run: yarn run docgen
         working-directory: packages/contracts/
       - name: Format with prettier
-        run: yarn run prettier 'packages/contracts/docs/osx/03-reference-guide/**/*.md' --write
+        run: yarn run prettier 'packages/contracts/docs/develper-portal/03-reference-guide/**/*.md' --write
```
Comment (Michael-A-Heuer) [inline .github/workflows/documentation-update.yml:29]:
```suggestion
        run: yarn run prettier 'packages/contracts/docs/developer-portal/03-reference-guide/**/*.md' --write
```
Please fix the folder name and paths everywhere

**Review by Michael-A-Heuer — CHANGES_REQUESTED** [review]
Please fix the folder name typo and deleted file

File: .github/workflows/documentation-update.yml (hunk 2 above)
Comment (mathewmeconry) [reply .github/workflows/documentation-update.yml]: done with af5827b4b0021a9d7320c9a9a4bdf7e5bdd71a88

File: packages/contracts/docs/osx/03-reference-guide/index.md (hunk 1 above)
Comment (mathewmeconry) [reply packages/contracts/docs/osx/03-reference-guide/index.md]: good catch. 
fixed with d5b3530ca42fa04fe50297fdad31015479adf0a3

**Review by Michael-A-Heuer — DISMISSED** [review]

**Review by Rekard0 — DISMISSED** [review]

**Pull request merged**